sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/JsonToRow.java

```diff
@@ -98,7 +98,7 @@ public void processElement(ProcessContext context) {
                       context.output(jsonToRow(objectMapper(), context.element()));
                     }
                   }))
-          .setCoder(schema.getRowCoder());
+          .setSchema(schema, SerializableFunctions.identity(), SerializableFunctions.identity());
```

Comment:
+1, some shorthand version would be helpful. Maybe default to `identity()` and have a couple of overloads of `setSchema()` to allow customization? Or wire it up to schema registry and default to `identity()` there? 

And is `setSchema` the right place to specify these transforms? Isn't it just conflating  `.apply(ToRow.withSchema())... .apply(ParDo.of(fromRow()))`? My thought is that if the transforms are non-trivial, then they are probably better be specified as `ParDos`

Reply:
These functions are not being applied here. All schemas in Beam must be registered with a conversion function to/from Row. These conversions are applied by Beam at the appropriate places (e.g. in the DoFn state machine). They can't really be specified as part of ParDo, because they are needed to implement ParDo behind the scenes!

In this case, it's clearly not needed as the type is already Row. Will add a simpler function for PCollection<Row>